packages/dx-react-scheduler-material-ui/src/templates/views/horizontal/time-table/layout.jsx

```diff
@@ -12,49 +12,84 @@ const styles = {
   },
 };
 
-const LayoutBase = ({
-  cellComponent: Cell,
-  rowComponent: Row,
-  classes,
-  tableRef,
-  className,
-  cellsData,
-  formatDate,
-  ...restProps
-}) => (
-  <RootRef rootRef={tableRef}>
-    <TableMUI
-      className={classNames(classes.table, className)}
-      {...restProps}
-    >
-      <TableBody>
-        {cellsData.map(row => (
-          <Row key={row[0].startDate.toString()}>
-            {row.map(({
-              startDate,
-              endDate,
-              today,
-              otherMonth,
-            }) => (
-              <Cell
-                key={startDate}
-                startDate={startDate}
-                endDate={endDate}
-                today={today}
-                otherMonth={otherMonth}
-                formatDate={formatDate}
-              />
+class LayoutBase extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.table = null;
+
+    this.saveReference = this.saveReference.bind(this);
+  }
+
+  componentDidMount() {
+    const { setCellElements } = this.props;
+
+    const cellElements = this.table.querySelectorAll('td');
+    setCellElements(cellElements);
+  }
+
+  componentDidUpdate() {
+    const { setCellElements } = this.props;
+
+    const cellElements = this.table.querySelectorAll('td');
+    setCellElements(cellElements);
+  }
+
+  saveReference(ref) {
+    const { tableRef } = this.props;
+    this.table = ref;
+    tableRef(ref);
+  }
+
+  render() {
+    const {
+      cellComponent: Cell,
+      rowComponent: Row,
+      classes,
+      tableRef,
+      className,
+      cellsData,
+      formatDate,
+      setCellElements,
+      ...restProps
+    } = this.props;
+    return (
+      <RootRef rootRef={this.saveReference}>
```

Comment:
Can we extract `RootRef -> Table -> Body` to a container component to be used in both horizontal and vertical layouts?

Reply:
No, because our Scheduler consists of 2/3 (horizontal/vertical) tables.

Reply:
This container component is supposed to be rendered in each view as it is now. I suggest to extract this refs logic:
`RootRef -> Table -> Body -> Row`
 to
`LayoutContainer -> Row`